examples/Tests/EQDiagram.hpp

```diff
@@ -0,0 +1,267 @@
+    It needs two values to properly run:
+        - The number of plot values to compute (N) which is passed as a template argument;
+        - The sample rate (sampleRate), needed to normalize frequencies when computing a filter's response to a certain frequency, which should be set before use of the object.
+*/
+template <int N, int bandNb>
+struct EQDiagramCalculator
+{
+  EQDiagramCalculator()
+      : sampleRate{44100.f}
+  {
+    auto eps = (range::max - range::min) / (N - 1);
+    for (auto i = 0; i < N; ++i)
+      frequencies.push_back(range::min + eps * i);
+    filters.resize(bandNb);
+  }
+
+  struct range
+  {
+    static const int plotAmount = N;
+    static const double min = 20.f;
+    static const double max = 20000.f;
+  };
+
+  float sampleRate;
+
+  std::vector<float> frequencies{}; //abscissa values
+  std::vector<float> responses{};   //ordinate values
+
+  using FilterHolder = Dsp::Filter;
+  std::vector<const FilterHolder*> filters{};
+
+  void addFilter(int band, const FilterHolder* filter)
+  {
+    if (filters[band] != filter)
+    {
+      filters[band] = filter;
+    }
+  }
+
+  bool removeFilter(int band) 
+  { 
+    if(filters[band] == nullptr)
+      return false;
+    filters[band] = nullptr;
+    return true;
+  }
+  /*
+        We consider each filter to be connected in series. In this case, the resulting transfer function is the 
+        multiplication of each filter's transfer function and the gain response in the sum of the norms of 
+        each filter's complex responses.
+    */
+  void computeDiagram()
+  {
+    // we erase previous values contained in responses as there might be new filters added / removed filters since last computation
+    responses = std::vector<float>();
+    for (auto& w : frequencies)
+    {
+      float res{};
+      for (auto& f : filters)
+      {
+        if (f)
+          res += log10(std::norm(f->response(w / sampleRate)));
+      }
+      responses.push_back(20. * res);
+    }
+  }
+};
+
+/* 
+    UI Object to display values computed by an EQDiagramCalculator. When initialized with halp::custom_item_base<EQDiagram>, xInput, yInput and valuesNb should be specified
+    to match an EQDiagramCalculator's frequencies, responses and plotAmount value.
+*/
+template <int w, int h, int N>
+struct EQDiagram
+{
+  EQDiagram()
+      : valuesNb(N)
+  {
+    //maybe should fill with coords to draw flatline at start
+    abscissa.resize(N);
+    ordinate.resize(N);
+  }
+  static constexpr double width() { return w; }
+  static constexpr double height() { return h; }
+  static constexpr double layout()
+  {
+    enum
+    {
+      custom
+    } d{};
+    return d;
+  }
+
+  // double x = 0.0;
+  // double y = 0.0;
+
+  int valuesNb;
+
+  // the followings will hold the coords of plotted points
+  std::vector<float> abscissa{};
+  std::vector<float> ordinate{};
+
+  std::vector<float> xticks{20, 500, 1, 2, 5, 10, 20};
```

Comment:
since it's a fixed set of values you can directly use an array of floats: 

      static constexpr float xticks[] {20, 500, 1, 2, 5, 10, 20};